util/Keplr.js

```diff
@@ -0,0 +1,138 @@
+          // The 'stake' button in Keplr extension will link to the webpage.
+          // walletUrlForStaking: "",
+          bip44: {
+            coinType: 118,
+          },
+          bech32Config: {
+            bech32PrefixAccAddr: network.addressPrefix,
+            bech32PrefixAccPub: network.addressPrefix + 'pub',
+            bech32PrefixValAddr: network.addressPrefix + 'valoper',
+            bech32PrefixValPub: network.addressPrefix + 'valoperpub',
+            bech32PrefixConsAddr: network.addressPrefix + 'valcons',
+            bech32PrefixConsPub: network.addressPrefix + 'valconspub',
+          },
+          currencies: network.coinLookup.map(({ viewDenom }) => configToKeplrCoin(viewDenom)),
+          feeCurrencies: network.coinLookup.map(({ viewDenom }) => configToKeplrCoin(viewDenom)),
+          // (Optional) The number of the coin type.
+          // This field is only used to fetch the address from ENS.
+          // Ideally, it is recommended to be the same with BIP44 path's coin type.
+          // However, some early chains may choose to use the Cosmos Hub BIP44 path of '118'.
+          // So, this is separated to support such chains.
+          coinType: 118,
+          // (Optional) This is used to set the fee of the transaction.
+          // If this field is not provided, Keplr extension will set the default gas price as (low: 0.01, average: 0.025, high: 0.04).
+          // Currently, Keplr doesn't support dynamic calculation of the gas prices based on on-chain data.
+          // Make sure that the gas prices are higher than the minimum gas prices accepted by chain validators and RPC/REST endpoint.
+          // gasPriceStep: {
+          //     low: 0.01,
+          //     average: 0.025,
+          //     high: 0.04
+          // }
+        });
+        await window.keplr.enable(network.id);
+
+        const offlineSigner = window.getOfflineSigner(network.id);
+        this.signer = offlineSigner;
+        this.accounts = await offlineSigner.getAccounts();
+        this.inited = true;
+        return true;
+      } catch (error) {
+        console.error(error);
+      }
+    }
+    return false;
+  }
+
+  async getKeplrSigner() {
+    if (!this.inited) {
+      const res = await this.initKeplr();
+      if (!res) throw new Error('CANNOT_INIT_KEPLR');
+    }
+    return this.signer;
+  }
+
+  getWalletAddress() {
+    const [wallet = {}] = this.accounts;
+    return wallet.address;
+  }
+
+  async signLogin(signPayload) {
+    if (!this.inited) {
+      const res = await this.initKeplr();
+      if (!res) throw new Error('CANNOT_INIT_KEPLR');
+    }
+    const message = {
+      chain_id: network.id,
+      memo: signPayload,
+      msgs: [],
+      fee: { gas: '1', amount: { denom: 'nanolike', amount: '0' } },
+    };
+    const payload = await this.signer.sign(
+      this.accounts[0].address,
```

Comment:
can replace with `this.getWalletAddress()`?

Reply:
Added a new method for getting address internally